include/shell/shell.h

```diff
@@ -36,6 +36,21 @@ extern "C" {
 
 #define SHELL_CMD_ROOT_LVL		(0u)
 
+/** @brief Flag indicates that optional arguments will be treated as one,
+ *	   unformatted argument.
+ *
+ * By default, shell is parsing all arguments, treats all spaces as argument
+ * separators unless they are within quotation marks which are removed in that
+ * case. If command rely on unformatted argument then this flag shall be used
+ * in place of number of optional arguments in command definition to indicate
+ * that only mandatory arguments shall be parsed and remaining command string is
+ * passed as a raw string.
+ */
+#define SHELL_OPT_ARG_RAW	(0xFE)
```

Comment:
I think that information about `SHELL_OPT_ARG_RAW` and `SHELL_ARG_MAX` values shall be added in KCONFIG help for `SHELL_ARGC_MAX`. Or maybe we could limit `SHELL_ARGC_MAX` in KCONFIG ?

Reply:
i think there is no relation between those 2. This is limit for optional arguments, there are also mandatory arguments so in theory `SHELL_ARGC_MAX` could be more than 255, even more than 510. Of course, this is rather unrealistic values so i don't want to put any relation there.